Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Build the pathfinder binary
+FROM rust:1.59-bullseye AS builder
+
+WORKDIR /usr/src/pathfinder
+COPY . .
+
+# Run pathfinder in a slimmer image
+RUN cargo build --release --bin pathfinder
+
+FROM debian:bullseye-slim AS runner
+
+RUN DEBIAN_FRONTEND=noninteractive \
+    && apt-get update \
+    && apt-get install -y ca-certificates tzdata libgmp-dev python3 python3-pip \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=builder /usr/src/pathfinder/target/release/pathfinder /usr/local/bin/pathfinder
+
+WORKDIR /usr/share/pathfinder
+COPY py py
+RUN python3 -m pip install --upgrade pip
+RUN python3 -m pip install -r py/requirements-dev.txt
+
+# Pathfinder expects a very specific directory structure for the
+# python environment and scripts
+# This docker image is not using virtual environments, so replicate
+# the directory structure with a shim python script.
+ENV VIRTUAL_ENV=/usr/share/pathfinder/py/venv
+RUN mkdir -p ${VIRTUAL_ENV}/bin
+RUN echo '#!/bin/bash\nexec "/usr/bin/python3" "$@"' >> ${VIRTUAL_ENV}/bin/python
+RUN chmod +x ${VIRTUAL_ENV}/bin/python
+RUN ${VIRTUAL_ENV}/bin/python --version
+
+# Create directory and volume for persistent data
+RUN mkdir -p /usr/share/pathfinder/data
+
```

Comment:
We shouldn't run this container as root. It would be good to toss a `USER 1000` in here and ensure all the right things still work.

Reply:
You're right, I'll fix this.

Reply:
Thank you!

Reply:
@fracek I've tested your fixes and everything works for me :tada: . I'l retest when the latest touches are in (ie. running as non-root).